backend/server/src/db/migrations/3-realms.sql

```diff
@@ -28,8 +31,98 @@ select setval(
     xtea(0, (select key from __xtea_keys where entity = 'realm'), false),
     false
 );
-insert into realms (name, parent, path) values ('', null, '');
+insert into realms (name, parent, path_segment, full_path) values ('', null, '', '');
+
+
+
+-- Triggers to update `full_path` ---------------------------------------------------------
+--
+-- The `full_path` column is completely managed by triggers to always have the
+-- correct values according to `path_segment` and `parent`. Doing this is a bit
+-- tricky.
+
+-- The `before insert` trigger is straight forward because we know that there
+-- don't exist any children yet that have to be updated. We make sure that
+-- insert operations do not try to specifcy the `full_path` already since that
+-- would be overwritten anyway.
+create function set_full_realm_path() returns trigger as $$
+begin
+    if NEW.full_path is not null then
+        raise exception 'do not set the full path of a realm directly (for realm %)', NEW.id;
+    end if;
+
+    NEW.full_path := (select full_path from realms where id = NEW.parent) || '/' || NEW.path_segment;
+    return NEW;
+end;
+$$ language plpgsql;
+
+-- However, handling updates gets interesting since we potentially have to
+-- update a large number of (indirect) children. To visit all descendents of a
+-- realm, we could have a recursive function, for example. But: changing those
+-- descendents via `update` would cause the another trigger to get triggered! I
+-- don't think one can avoid that. But we can just use this to our advantage
```

Comment:
I think we could avoid that. Something something

```sql
CREATE TRIGGER set_history
BEFORE UPDATE ON field_data
FOR EACH ROW 
WHEN (pg_trigger_depth() < 1)
EXECUTE PROCEDURE gener_history();
```

Source: https://dba.stackexchange.com/a/103661

Reply:
Mhhh interesting. Do you think a solution like that has an advantage over the current solution?